cvat/apps/analytics_report/report/create.py

```diff
@@ -0,0 +1,319 @@
+            next_update_time = last_update_time + delay * intervals
+            queue_job_id = self._make_regular_queue_job_id(obj, next_update_time)
+
+        return queue_job_id
+
+    class AnalyticsReportsNotAvailable(Exception):
+        pass
+
+    def schedule_analytics_report_autoupdate_job(self, *, job=None, task=None, project=None):
+        now = timezone.now()
+        delay = self._get_analytics_check_job_delay()
+        next_job_time = now.utcnow() + delay
+
+        scheduler = self._get_scheduler()
+        existing_job_ids = set(j.id for j in scheduler.get_jobs(until=next_job_time))
+
+        target_obj = None
+        cvat_project_id = None
+        cvat_task_id = None
+        if job is not None:
+            if job.segment.task.project:
+                target_obj = job.segment.task.project
+                cvat_project_id = target_obj.id
+            else:
+                target_obj = job.segment.task
+                cvat_task_id = target_obj.id
+        elif task is not None:
+            if task.project:
+                target_obj = task.project
+                cvat_project_id = target_obj.id
+            else:
+                target_obj = task
+                cvat_task_id = target_obj.id
+        elif project is not None:
+            target_obj = project
+            cvat_project_id = project.id
+
+        queue_job_id = self._find_next_job_id(existing_job_ids, target_obj, now=now)
+        if queue_job_id not in existing_job_ids:
+            scheduler.enqueue_at(
+                next_job_time,
+                self._check_job_analytics,
+                cvat_task_id=cvat_task_id,
+                cvat_project_id=cvat_project_id,
+                job_id=queue_job_id,
+            )
+
+    def schedule_analytics_check_job(self, *, job=None, task=None, project=None, user_id):
+        rq_id = self._make_custom_analytics_check_job_id()
+
+        queue = self._get_queue()
+        queue.enqueue(
+            self._check_job_analytics,
+            cvat_job_id=job.id if job is not None else None,
+            cvat_task_id=task.id if task is not None else None,
+            cvat_project_id=project.id if project is not None else None,
+            job_id=rq_id,
+            meta={"user_id": user_id, "job_type": self._RQ_CUSTOM_ANALYTICS_CHECK_JOB_TYPE},
+            result_ttl=self._JOB_RESULT_TTL,
+            failure_ttl=self._JOB_RESULT_TTL,
+        )
+
+        return rq_id
+
+    def get_analytics_check_job(self, rq_id: str):
+        queue = self._get_queue()
+        rq_job = queue.fetch_job(rq_id)
+
+        if rq_job and not self.is_custom_analytics_check_job(rq_job):
+            rq_job = None
+
+        return rq_job
+
+    def is_custom_analytics_check_job(self, rq_job) -> bool:
+        return rq_job.meta.get("job_type") == self._RQ_CUSTOM_ANALYTICS_CHECK_JOB_TYPE
+
+    @classmethod
+    def _check_job_analytics(cls, *, cvat_job_id: int=None, cvat_task_id: int=None, cvat_project_id: int=None) -> int:
```

Comment:
Probably, a better name for this function (and the class, actually) should be chosen, because it mentions "job" in the name, while computes reports for tasks and projects as well.